### ISSUE TITLE
Unable to startup CSI provisioners

### ISSUE THREAD

**Issue by narphu** [kadalu/kadalu#324]
I am a beginner and using flannel CNI deployed with kubespray as the k8s base. I was trying to deploy the kadalu operaor but this the logs I see in the operator:
+ pid=0
+ cmd=/usr/bin/python3
+ script=/kadalu/main.py
+ trap 'kill ${!}; term_handler' SIGTERM
+ pid=6
+ true
+ wait 7
+ /usr/bin/python3 /kadalu/main.py
+ tail -f /dev/null
[2020-10-22 17:09:11,207] WARNING [connectionpool - 750:urlopen] - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f8a97ebcf50>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v1/namespaces/kadalu/configmaps
[2020-10-22 17:09:11,207] WARNING [connectionpool - 750:urlopen] - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f8a968a1290>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v1/namespaces/kadalu/configmaps
[2020-10-22 17:09:11,208] WARNING [connectionpool - 750:urlopen] - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f8a968a1310>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v1/namespaces/kadalu/configmaps
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 160, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 84, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 74, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

Not sure if this something specific to flannel or the way I have setup the networking. Everything is default kubespray.

**Comment by narphu** [comment]
As a follow up, I observed the following in the kube-controller-manager pods:
` 1 resource_quota_controller.go:437] failed to sync resource monitors: [couldn't start monitor for resource "kadalu-operator.storage/v1alpha1, Resource=kadalustorages": unable to monitor quota for resource "kadalu-operator.storage/v1alpha1, Resource=kadalustorages", couldn't start monitor for resource "extensions/v1beta1, Resource=networkpolicies": unable to monitor quota for resource "extensions/v1beta1, Resource=networkpolicies"]`

**Comment by chernetskyi** [comment]
Same on AWS EKS (AWS VPC CNI plugin) with self-managed nodes.

Upd: tried to fix the issue as mentioned in #165 - no luck.

**Comment by aravindavk** [comment]
Identified an issue with latest version of Python kubernetes Client. Still looking at identifying the root cause of the failure.

CI is passing with the downgraded version of python client. https://github.com/kadalu/kadalu/pull/325

To confirm the issue,

* Login to operator pod, Example: `kubectl exec -it operator-76b85b86b6-cn2tx -n kadalu -- /bin/bash`
* Run `pip3 install -Iv kubernetes==11.0.0`
* Start Operator service by running `python3 /kadalu/main.py`(inside Operator Pod)
* Open another terminal and see if Kadalu Operator starts other pods(csi pods)

**Comment by chernetskyi** [comment]
Cannot create storage pool with `kubectl kadalu storage-add` after downgrading Python kubernetes in the Operator Pod.

```python3
Traceback (most recent call last):
  File "/kadalu/main.py", line 555, in <module>
    main()
  File "/kadalu/main.py", line 541, in main
    deploy_storage_class()
  File "/kadalu/main.py", line 517, in deploy_storage_class
    template(filename, namespace=NAMESPACE, kadalu_version=VERSION)
  File "/kadalu/main.py", line 38, in template
    with open(filename + ".j2") as template_file:
FileNotFoundError: [Errno 2] No such file or directory: '/kadalu/templates/storageclass-kadalu.replica3.yaml.yaml.j2'
```

**Comment by amarts** [comment]
@aravindavk fixed the issue through #325, which is merged now. Did the issue get resolved with it?

**Comment by chernetskyi** [comment]
Yes, it is now resolved.
